Generators/share/external/pythia6.C

```diff
@@ -0,0 +1,37 @@
+#include "TSystem.h"
+#include "TPythia6.h"
+
+double momentumUnit = 1.;            // [GeV/c]
+double energyUnit   = 1.;            // [GeV/c]
+double positionUnit = 0.1;           // [cm]
+double timeUnit     = 3.3356410e-12; // [s]
+
+void configure(TPythia6 *py6, const char *params);
+
+TGenerator *
+pythia6(double energy = 14000., const char *params = nullptr)
+{
+  gSystem->Load("libpythia6");
+  auto py6 = TPythia6::Instance();
+  if (params) configure(py6, params);
+  py6->Initialize("CMS", "p", "p", energy);
+  return py6;
+}
+
+void
+configure(TPythia6 *py6, const char *params)
+{
+  std::ifstream file(params);
+  if (!file.is_open()) {
+    std::cerr << "Cannot open configuration file: " << params << std::endl;
+    return;
+  };
+  std::string line, command;
+  while (std::getline(file, line)) {
+    /** remove comments **/
+    command = line.substr(0, line.find_first_of("#"));
+    if (command.length() == 0) continue;
+    py6->Pygive(command.c_str());
+  }
+  file.close();
+}
```

Comment:
could you add some docu on what are the constraints (template) for such macro files?
Another example interfacing with an AliRoot generator might also be useful.

Reply:
I have added `share/Generators/external/extgen.C` as a template function with information about the constraints for the macro files.

I have added `share/Generators/external/hijing.C` as an example to interface with `AliGenHijing` generator of AliRoot